### ISSUE TITLE
Reproduction lines are not captured in test output

### ISSUE THREAD

**Issue by gwbrown** [elastic/elasticsearch#72818]
The test `NodeShutdownIT.testCRUD` failures (see https://github.com/elastic/elasticsearch/issues/72787) do not include repro lines with the failures.

Example build scans: [One](https://gradle-enterprise.elastic.co/s/zzvdmnhxpoudm/tests/:x-pack:plugin:shutdown:qa:multi-node:javaRestTest/org.elasticsearch.xpack.shutdown.NodeShutdownIT/testCRUD?top-execution=1), [Two](https://gradle-enterprise.elastic.co/s/gy32ik4jax3ua/tests/:x-pack:plugin:shutdown:qa:multi-node:javaRestTest/org.elasticsearch.xpack.shutdown.NodeShutdownIT/testCRUD?top-execution=1)

This appears to happen reliably with this test.

PR https://github.com/elastic/elasticsearch/pull/72772 should fix the test failures, which occur because the other test in `NodeShutdownIT`, `testAllocationPreventedForRemoval`, does not remove the node shutdown it registers, which leaks into `testCRUD` and causes assumptions to fail. The failure will occur any time `testAllocationPreventedForRemoval` runs before `testCRUD`. A repro line that appears to reproduce this reliably is:
```
./gradlew ':x-pack:plugin:shutdown:qa:multi-node:javaRestTest' --tests "org.elasticsearch.xpack.shutdown.NodeShutdownIT" -Dtests.seed=91EA8B18627FA3AA -Dtests.locale=ja -Dtests.timezone=America/Louisville
```

At some point, that repro line came from a `testCRUD` failure (it's been edited to run all of `NodeShutdownIT`), but I think it was a local failure? I've unfortunately lost my terminal history back that far.

**Comment by mark-vieira** [comment]
@breskeby My suspicion is the stuff around test retrying might be involved here. The reproduction lines are generated by a JUnit `RunListener` so it's possible the lifecycle on those events is slightly different than "normal" test case output and perhaps those are getting dropped by our rerun results processor. Could you look into this?

**Comment by mark-vieira** [comment]
To clarify, others have reported this as well so I don't think it's specific to the test case Gordon reported. From what I've observed, this is happening across the board.

**Comment by breskeby** [comment]
I've raised #72831 to disable the rerun functionality till we sorted out the issues we are seeing. this should also fix this issue here

**Comment by mark-vieira** [comment]
Closing this since it should be addressed by #72831.
